### PR TITLE
Replace ContainsKey and [] with TryGetValue for Dictionary

### PR DESCRIPTION
**Pull request by DamirAinullin**
Replace ContainsKey and [] with TryGetValue for Dictionary, some additional code formatting

### CODE REVIEW

**Comment by brianrob** [comment]
@DamirAinullin, thank you for your contribution.  This change is reasonably large, and touches many different places, some of which I think might benefit from this, but others I suspect we wouldn't notice.  Do you have a scenario in mind where this makes a material performance difference?

**Comment by DamirAinullin** [comment]
@brianrob, no, I just went through whole solution.

File: src/PerfView/HeapView/HeapAnalyzer.cs [hunk 1]
```diff
@@ -933,12 +932,11 @@ public bool LoadEvents(int procID, int sampleInterval100ns)
             }
 
             // Get the process we want
-            if (!m_processLookup.ContainsKey(procID))
+            if (!m_processLookup.TryGetValue(procID, out m_process))
```
Comment (sharwell) [inline src/PerfView/HeapView/HeapAnalyzer.cs:935]: 📝 This one appears to be a behavior change if `m_process` is not null when this line is reached.

File: src/PerfView/OverweightReport.cs [hunk 2]
```diff
@@ -165,13 +165,12 @@ private class Result
             foreach (var key in d1.Keys)
```
Comment (sharwell) [inline src/PerfView/OverweightReport.cs:165]:
```suggestion
            foreach (var (key, v1) in d1)
```

File: src/PerfView/OverweightReport.cs [hunk 3]
```diff
@@ -165,13 +165,12 @@ private class Result
...
             {
                 // skip symbols that are not in both traces
-                if (!d2.ContainsKey(key))
+                if (!d2.TryGetValue(key, out var v2))
                 {
                     continue;
                 }
 
                 var v1 = d1[key];
```
Comment (sharwell) [inline src/PerfView/OverweightReport.cs:173]:
```suggestion
```

**Review by sharwell — COMMENTED** [review]
I've made it part way through. I agree with the primary change in principle, I'm seeing a whole bunch of unrelated formatting changes which is slowing down the review.

**Comment by DamirAinullin** [comment]
@sharwell agree with yout point about formatting, sorry about that. It was just auto-formatting from VisualStudio.

**Pull request closed without merging**